_posts/2016-02-25-usage-tests.md

```diff
@@ -0,0 +1,73 @@
+---
+layout: post
+title: Usage Tests
+---
+
+Tests suites have a number of purposes.
+Perhaps most commonly you think of the test suite as validating the behavior of your code.
+Absolutely.
+But they also serve as examples of using the lib.
+Let's make that more explicit.
+
+## Usage
+
+If you have perused the test suites for library code, you may have noticed some conventions.
+Folks tend to have a `units/` directory for unit (read: isolated) tests.
+Other places you may see a `lib/` directory if the author is trying to keep the file to file organization the same between their implementation and tests.
+Finally you may have seen an `integration/`, `acceptance/`, etc. directory that indicates tests of various pieces working together.
+Sometimes the integration directory (whatever it be called) also has tests illustrating example usage of the lib.
+
+Here's the thing.
+Probably all non-trivial, well organized code will have interfaces that aren't meant to be used publicly.
+Within a class we often mark methods as `private` to discourage external use.
+You may even do the same thing [with internal classes][private-class].
+
+Hard as you try, you will always have publicly _accessible_ interfaces that were never intended for external use.
+Recently, I was [working on ActionCable][ac-pr] and a [suggestion] was made that we move the change closer to the library's user interface.
+It became clear with that suggestion how difficult it is to discern what portion of ActionCable was intended to be used directly.
+It's actually quite small.
+
+So how do you communicate this to the reader.
+Perhaps by organizing your test suite better.
```

Comment:
Something about the word "perhaps" here made me read it as though you were just idly speculating. Really you're strongly suggesting: "let's talk about one possibility: organizing your test suite better." Some kind of different phrasing I think would draw my attention more to the fact of "OK this is where he's going with this."